test/integration/emulator/FakeVideoDecoder.h

```diff
@@ -0,0 +1,244 @@
+#pragma once
+#include "test/bwe/FakeVideoSource.h"
+#include <algorithm>
+#include <list>
+#include <vector>
+
+#define DEBUG_FAKE_VIDEO_DECODER 0
+
+#if DEBUG_FAKE_VIDEO_DECODER
+#define FAKE_VIDEO_DECODER_LOG(fmt, ...) logger::debug(fmt, ##__VA_ARGS__)
+#else
+#define FAKE_VIDEO_DECODER_LOG(fmt, ...)
+#endif
+
+namespace emulator
+{
+class FakeVideoDecoder
```

Comment:
I think this large class deserves its own cpp file

Reply:
Agree.